lahja/endpoint.py

```diff
@@ -550,6 +527,68 @@ def _process_item(self, item: BaseEvent, config: Optional[BroadcastConfig]) -> N
             for handler in self._handler[event_type]:
                 handler(item)
 
+    async def _who_are_you_stream_handler(self) -> None:
+        """
+        Long running process that responds to _WhoAreYou events which are
+        typically send from server to client to establish a reverse connection.
+        """
+        self.logger.debug('Endpoint %s running _WhoAreYou mesage processor', self.name)
+        self._who_are_you_stream_handler_running.set()
+        async for request in self.stream(_WhoAreYou):
+            await self.broadcast(_MyNameIs(self.name), request.response_config())
+
+    async def _establish_reverse_connection(self, remote: RemoteEndpoint) -> None:
+        item = _WhoAreYou()
+        # We bind the event as if it was a request/response, giving it an _id
+        # which allows the remote to respond to us without needing to receive a
+        # subscription update.
+        item.bind(self, str(uuid.uuid4()))
```

Comment:
TODO: profile and optimize if we can gain some speed by using a different source of unique ids.

Reply:
I believe a scheme such as `endpoint_name` + timestamp + continuous integer should do it. The timestamp would be to get some extra safety. E.g. if we ever do what @lithp once suggested that we would save events into an internal database so that we could replay them, then just going with `endpoint_name` + continuous integer would probably not be enough to guarantee uniqueness.